### PR TITLE
Update scratch org definition files to use orgPreferenceSettings

### PR DESCRIPTION
**Pull request by davidmreed**
# Critical Changes

# Changes

 - Reformat scratch org definition files to use [`orgPreferenceSettings` instead of `orgPreferences`](https://developer.salesforce.com/docs/atlas.en-us.sfdx_dev.meta/sfdx_dev/sfdx_dev_scratch_orgs_def_file_config_values.htm), which is now deprecated.
 - Parameterize generation of scratch org definition files based on managed or unmanaged environments.

# Issues Closed

### CODE REVIEW

File: orgs/release.json [hunk 1]
```diff
@@ -1,11 +1,11 @@
...
-    "enabled": [
-      "S1DesktopEnabled",
-      "Translation",
-      "ChatterEnabled" 
-    ]
+  "settings": {
+    "orgPreferenceSettings": {
+      "s1DesktopEnabled": true,
+      "chatterEnabled": true,
+      "translation": true
```
Comment (davisagli) [inline orgs/release.json]: While we're working here, let's disable translation for the beta and release scratch orgs. It refers to translation workbench which is only needed for development. We had one bug where code was written that depended on translation workbench being installed (by referring to a permission that only exists in that case) and we didn't catch it until installing in an org with translation workbench turned off.

File: cumulusci/files/templates/project/scratch_def.json [hunk 2]
```diff
@@ -1,10 +1,11 @@
...
-  "orgPreferences" : {
-    "enabled": [
-      "S1DesktopEnabled",
-      "ChatterEnabled"
-    ]
+  "settings": {
+    "orgPreferenceSettings": {
+      "s1DesktopEnabled": true,
+      "chatterEnabled": true,
+      "translation": true
```
Comment (davisagli) [inline cumulusci/files/templates/project/scratch_def.json]: See my other comment -- this means that we want translation on for some scratch orgs but not others. Which means we either need two different template files, or a jinja2 conditional within the template based on a variable that we pass when rendering it.

Comment (jlantz) [reply cumulusci/files/templates/project/scratch_def.json]: I'd say the variable should be to signify an unmanaged vs a managed environment.  The dev and feature org shapes are unmanaged, beta and release are managed.

**Comment by davidmreed** [comment]
@davisagli / @jlantz If the updates look good to you I will re-create this pull request on a branch within the main repo, rather than my fork.

File: cumulusci/files/templates/project/scratch_def.json [hunk 3]
```diff
@@ -1,10 +1,13 @@
...
-    "enabled": [
-      "S1DesktopEnabled",
-      "ChatterEnabled"
-    ]
+  "settings": {
+    "orgPreferenceSettings": {
+      "s1DesktopEnabled": true
+      , "chatterEnabled": true
+      {% if not managed %}
+      , "translation": true
```
Comment (davisagli) [inline cumulusci/files/templates/project/scratch_def.json]: I'm not a fan of the commas at the start of the line, but I see how it helps with the conditional and I'm willing to put up with it.

**Comment by davisagli** [comment]
@davidmreed let's get this opened as a PR from our github org so we can see what the builds say.

File: cumulusci/files/templates/project/scratch_def.json (hunk 3 above)
Comment (davidmreed) [reply cumulusci/files/templates/project/scratch_def.json]: I don't love it either but it seemed less messy than the alternative conditional style.

**Pull request closed without merging**